sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3547,6 +3547,23 @@ def post_teardown_cleanup(self,
                             cluster_name=handle.cluster_name,
                             stdout=tpu_stdout,
                             stderr=tpu_stderr))
+        if handle.launched_resources.is_image_managed and terminate:
+            # Delete the image when terminating the cluster and the image is
+            # created by SkyPilot (--clone-disk-from)
+            logger.debug(f'Deleting image {handle.launched_resources.image_id}')
+            cluster_resources = handle.launched_resources
+            cluster_cloud = cluster_resources.cloud
+            image_dict = cluster_resources.image_id
+            assert cluster_cloud is not None, cluster_resources
+            assert image_dict is not None and len(image_dict) == 1
+            image_id = list(image_dict.values())[0]
+            try:
+                cluster_cloud.delete_image(image_id,
+                                           handle.launched_resources.region)
+            except exceptions.CommandError as e:
+                logger.debug(
```

Comment:
warn? "Failed to delete cloned image ... Please remove it manually to avoid image leakage. Details: ..."